pkg/codegen/internal/test/testdata/simple-resource-schema/schema.json

```diff
@@ -71,12 +71,14 @@
     "example::Resource": {
       "properties": {
         "bar": {
-          "type": "string"
+          "type": "string",
+          "secret": true
         }
       },
       "inputProperties": {
         "bar": {
-          "type": "string"
+          "type": "string",
+          "secret": true
```

Comment:
What is the requirement for this to work?  Will the input be marked as secret if only `properties` is marked secret? If only `inputProperties` is marked secret, will that do anything (it looked like perhaps the codegen doesn't pick that up?). Might be worth test coverage for the supported cases?

Reply:
If the property is marked secret, that will set the `additionalSecretOutputs` field. If the inputProperty is marked secret, it will use `pulumi.secret` to encrypt the input.

> Might be worth test coverage for the supported cases?
I tested this change with the codegen tests in this repo as well as integration tests in the k8s provider.